improver_tests/categorical/decision_tree/test_ModalFromGroupings.py

```diff
@@ -279,6 +311,21 @@ def test_expected_values_day_weighting(
 @pytest.mark.parametrize(
     "data, ignore_intensity, expected, reverse_intensity_dict",
     (
+        # Dry dominated. Rain contributes overcast conditions. Cloud therefore
+        # becomes the dry code. The cloud group contains 2 overcast codes and
+        # 1 cloudy code, so overcast (8) is chosen.
+        ([1, 1, 1, 7, 12, 12], True, 8, False),
+        # Dry dominated. Showers contribute partly cloudy conditions meaning
+        # the sunny/partly cloudy group provides the summary code. Of this
+        # there are 2 sunny and 2 partly cloud codes, with the latter chosen (3)
+        # as more significant.
+        ([1, 1, 10, 14, 8, 8, 8], True, 3, False),
+        # Dry. Partly cloudy and sunny are grouped by the intensity
+        # categorisation allowing them to dominate over the overcast
+        # codes. A partly cloud code is returned.
+        # cloud cover in the light rain codes is included in determining the
+        # dominant dry code, leading to an overcast symbol overall (8).
```

Comment:
This description doesn't seem to make sense, given that the result from the test seems to be 1 (sunny), rather than overcast.

Reply:
Yes, I must have been adding and deleting tests and fully mangled this one. Fixed.